eng/scripts/build.ps1

```diff
@@ -3,70 +3,91 @@ param([string]$filter, [switch]$clean, [switch]$vet, [switch]$generate, [switch]
 
 . (Join-Path $PSScriptRoot .. common scripts common.ps1)
 
-function Process-Sdk () {
+function Process-Sdk ()
+{
     $currentDirectory = Get-Location
-    if ($clean) {
+    if ($clean)
+    {
         Write-Host "##[command]Executing go clean -v ./... in " $currentDirectory
         go clean -v ./...
         if ($LASTEXITCODE) { exit $LASTEXITCODE }
     }
 
-    if ($cleanGenerated) {
+    if ($cleanGenerated)
+    {
         Write-Host "##[command]Cleaning auto-generated files in" $currentDirectory
         Remove-Item "zz_generated_*"
     }
 
-    if ($generate) {
+    if ($generate)
+    {
         Write-Host "##[command]Executing autorest.go in " $currentDirectory
         $autorestPath = "./" + $config
 
         $autorestVersion = "@autorest/go@4.0.0-preview.27"
-        if ($outputFolder -eq '') {
+        if ($outputFolder -eq '')
+        {
             $outputFolder = $currentDirectory
         }
         autorest --use=$autorestVersion --go --track2 --output-folder=$outputFolder --file-prefix="zz_generated_" --clear-output-folder=false $autorestPath
-        if ($LASTEXITCODE) {
+        if ($LASTEXITCODE)
+        {
             Write-Host "##[error]Error running autorest.go"
             exit $LASTEXITCODE
         }
     }
 
-    if ($format) {
+    if ($format)
+    {
         Write-Host "##[command]Executing gofmt -s -w . in " $currentDirectory
         gofmt -s -w .
-        if ($LASTEXITCODE) {exit $LASTEXITCODE}
+        if ($LASTEXITCODE) { exit $LASTEXITCODE }
     }
 
-    if ($tidy) {
+    if ($tidy)
+    {
         Write-Host "##[command]Executing go mod tidy in " $currentDirectory
         go mod tidy
-        if ($LASTEXITCODE) {exit $LASTEXITCODE}
+        if ($LASTEXITCODE) { exit $LASTEXITCODE }
     }
 
-    if (!$skipBuild) {
+    if (!$skipBuild)
+    {
         Write-Host "##[command]Executing go build -x -v ./... in " $currentDirectory
         go build -x -v ./...
         Write-Host "##[command]Build Complete!"
-        if ($LASTEXITCODE) {exit $LASTEXITCODE}
+        if ($LASTEXITCODE) { exit $LASTEXITCODE }
     }
 
-    if ($vet) {
+    if ($vet)
+    {
         Write-Host "##[command]Executing go vet ./... in " $currentDirectory
         go vet ./...
     }
 }
 
-try {
+try
+{
     $startingDirectory = Get-Location
 
     $sdks = Get-AllPackageInfoFromRepo $filter
 
-    foreach ($sdk in $sdks) {
+    foreach ($sdk in $sdks)
+    {
         Push-Location $sdk.DirectoryPath
         Process-Sdk
         Pop-Location
     }
+
+    if ($sdks.Count -eq 0 -and $filter)
+    {
+        Write-Host "Cannot find go module under $filter, considered to be an onboard RP package"
```

Comment:
Is this going to work in the general case? What if someone passes an invalid filter? Should we at least check to see if the location exists and if not then error in a different way instead of trying to process?

Reply:
Or are there any other flags that you are passing that might indicate that you are in an on-boarding process?

Reply:
Add config existing test.